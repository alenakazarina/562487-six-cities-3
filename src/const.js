export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

export const OfferType = {
  house: `House`,
  room: `Private room`,
  apartment: `Apartment`,
  hotel: `Hotel`
};

export const AppRoute = {
  ROOT: `/`,
  OFFER: `/offer/:id`,
  FAVORITES: `/favorites`,
  LOGIN: `/login`
};

export const RATINGS = [
  {
    value: 5,
    title: `perfect`
  },
  {
    value: 4,
    title: `good`
  },
  {
    value: 3,
    title: `not bad`
  },
  {
    value: 2,
    title: `badly`
  },
  {
    value: 1,
    title: `terribly`
  }
];

export const MONTHS = [
  `January`, `February`, `March`, `April`,
  `May`, `June`, `July`, `August`,
  `September`, `October`, `November`, `December`
];

export const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const InitialState = {
  USER: {
    authStatus: `NO_AUTH`,
    user: {
      id: -1,
      email: ``,
      name: ``,
      isPro: false,
      avatarUrl: ``
    }
  },
  FAVORITES: {
    favorites: []
  },
  OFFER: {
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  OFFERS: {
    offers: [],
    activeCity: ``
  },
  ERRORS: {
    errorStatus: null
  }
};

