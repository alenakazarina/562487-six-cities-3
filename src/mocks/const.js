const titles = [
  `Amazing and Extremely Central Flat`,
  `The house among olive`,
  `Nice, cozy, warm big bed apartment`
];

const galleryImages = [`image1`, `image2`, `image3`, `image4`, `image5`, `image6`, `image7`];

export const cities = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

const locations = [
  {
    name: `Paris`,
    location: {
      latitude: 1,
      longitude: 1,
      zoom: 13
    }
  },
  {
    name: `Dusseldorf`,
    location: {
      latitude: 2,
      longitude: 2,
      zoom: 13
    }
  }
];

const offerTypes = [`apartment`, `room`, `house`, `hotel`];

const goods = [`Breakfast`, `Laptop friendly workspace`, `Washer`];

const sentences = [
  `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  `In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`
];

export const users = [
  {
    avatarUrl: `/user1_avatar`,
    id: 1,
    isPro: true,
    name: `Alena`
  },
  {
    avatarUrl: `/user2_avatar`,
    id: 1,
    isPro: true,
    name: `Nick`
  }
];

export const appUsers = users.map((user) => Object.assign({}, user, {email: `user@gmail.com`}));

const dates = [
  new Date(`2020.03.04 14:00`),
  new Date(`2020.03.03 10:00`)
];

export const reviews = [{
  id: 1,
  user: users[0],
  rating: 4,
  comment: sentences[0],
  date: dates[0]
}, {
  id: 2,
  user: users[1],
  rating: 5,
  comment: sentences[1],
  date: dates[1]
}];

export const cityOffers = [{
  id: 1,
  city: locations[0],
  previewImage: galleryImages[0],
  images: galleryImages,
  title: titles[0],
  isFavorite: true,
  isPremium: true,
  rating: 5,
  type: offerTypes[0],
  bedrooms: 3,
  maxAdults: 5,
  price: 400,
  amenities: goods,
  host: users[0],
  description: sentences[0],
  location: locations[0].location
}, {
  id: 2,
  city: locations[1],
  previewImage: galleryImages[2],
  images: galleryImages,
  title: titles[1],
  isFavorite: false,
  isPremium: false,
  rating: 4,
  type: offerTypes[2],
  bedrooms: 1,
  maxAdults: 7,
  price: 500,
  amenities: goods,
  host: users[1],
  description: sentences[1],
  location: locations[1].location
}, {
  id: 3,
  city: locations[1],
  previewImage: galleryImages[2],
  images: galleryImages,
  title: titles[0],
  isFavorite: false,
  isPremium: true,
  rating: 3,
  type: offerTypes[1],
  bedrooms: 2,
  maxAdults: 2,
  price: 600,
  amenities: goods,
  host: users[0],
  description: sentences[0],
  location: locations[1].location
}];

export const favoriteOffers = cityOffers.map((offer) => Object.assign({}, offer, {isFavorite: true}));

export const apiMockOffers = [{
  'bedrooms': 3,
  'city': `Paris`,
  'description': sentences[1],
  'goods': goods,
  'host': {
    'avatar_url': `avatar`,
    'id': 1,
    'is_pro': true,
    'name': `Alena`
  },
  'id': 1,
  'images': galleryImages,
  'is_favorite': true,
  'is_premium': true,
  'location': locations[1],
  'max_adults': 3,
  'preview_image': galleryImages[0],
  'price': 600,
  'rating': 3,
  'title': titles[1],
  'type': offerTypes[1]
}];

export const apiMockReviews = [{
  'comment': sentences[1],
  'date': dates[1].toISOString(),
  'id': 1,
  'rating': 3.5,
  'user': {
    'avatar_url': `userAvatar`,
    'id': 1,
    'is_pro': true,
    'name': `Alena`
  }
}];
