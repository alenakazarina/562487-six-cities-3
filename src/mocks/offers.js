import {OffersData} from './const';
import {
  generateId,
  getImage,
  getRandomBoolean,
  getRandomNumber,
  getRandomFromArray,
} from './utils';

const generateOffer = () => {
  return {
    id: generateId(),
    previewImage: getImage(OffersData.images),
    title: getRandomFromArray(OffersData.titles),
    isFavorite: getRandomBoolean(),
    isPremium: getRandomBoolean(),
    rating: getRandomNumber({min: 0, max: OffersData.maxRating}),
    type: getRandomFromArray(OffersData.types),
    price: getRandomNumber({min: 0, max: 1000})
  };
};

export const generateOffers = (count = 3) => {
  return new Array(count).fill(``).map(generateOffer);
};

export const offers = [
  {
    city: `Paris`,
    offers: generateOffers(4)
  },
  {
    city: `Cologne`,
    offers: generateOffers(5)
  },
  {
    city: `Brussels`,
    offers: generateOffers(4)
  },
  {
    city: `Amsterdam`,
    offers: generateOffers(5)
  },
  {
    city: `Hamburg`,
    offers: generateOffers(4)
  },
  {
    city: `Dusseldorf`,
    offers: generateOffers(5)
  }
];
