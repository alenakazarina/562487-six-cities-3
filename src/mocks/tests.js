import configureStore from 'redux-mock-store';
import {AuthStatus} from '../reducers/user/user';
import {cityOffers, reviews, appUsers} from './const';

const defaultUser = {
  id: -1,
  email: ``,
  name: ``,
  isPro: false,
  avatarUrl: ``
};

const mockStore = configureStore([]);

export const storeWithAuth = mockStore({
  USER: {
    authStatus: AuthStatus.AUTH,
    user: appUsers[0]
  },
  OFFERS: {
    offers: cityOffers,
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: cityOffers[1],
    activeOffer: cityOffers[2],
    nearOffers: cityOffers.slice(0, 2),
    comments: reviews
  },
  FAVORITES: {
    favorites: cityOffers
  }
});

export const storeWithNoAuth = mockStore({
  USER: {
    authStatus: AuthStatus.NO_AUTH,
    user: defaultUser
  },
  OFFERS: {
    offers: cityOffers,
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: null,
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  FAVORITES: {
    favorites: []
  }
});

export const storeWithNoOffers = mockStore({
  USER: {
    authStatus: AuthStatus.NO_AUTH,
    user: defaultUser
  },
  OFFERS: {
    offers: [],
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: null,
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  FAVORITES: {
    favorites: []
  }
});
