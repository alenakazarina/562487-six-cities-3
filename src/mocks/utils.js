import nanoid from 'nanoid';
import {IMAGE_PATH, OffersData} from './const';

const getRandomNumber = ({min = 0, max = 1}) => {
  return min + Math.floor(max * Math.random());
};

const getRandomFromObject = (obj = {}) => {
  const randomKey = getRandomFromArray(Object.keys(obj));
  return obj[randomKey];
};

const getRandomFromArray = (arr = []) => {
  const randomIndex = getRandomNumber({min: 0, max: arr.length});
  return arr[randomIndex];
};

const getRandomArray = ({items = [], minLength = 0, maxLength = 1}) => {
  const length = getRandomNumber({min: minLength, max: maxLength + 1});
  return new Array(length).fill(``).map(() => {
    const index = getRandomNumber(0, items.length);
    return items[index];
  });
};

const getRandomBoolean = () => {
  return Math.random() > 0.5;
};

const generateId = () => {
  return nanoid();
};

const getImage = (items = []) => {
  const randomIndex = getRandomNumber({min: 0, max: items.length});
  return IMAGE_PATH.concat(items[randomIndex]);
};

const generateImages = () => {
  return OffersData.images.map((image) => IMAGE_PATH.concat(image));
};

const formatMonthYear = (date = new Date()) => {
  return `${date.getMonth() + 1} ${date.getFullYear()}`;
};

export {
  getRandomNumber,
  getRandomArray,
  getRandomBoolean,
  generateId,
  getImage,
  generateImages,
  getRandomFromArray,
  getRandomFromObject,
  formatMonthYear
};
