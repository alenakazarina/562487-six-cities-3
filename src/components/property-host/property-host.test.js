import React from 'react';
import renderer from 'react-test-renderer';
import PropertyHost from './property-host';
import {USERS} from '../../mocks/const';

const CHILDREN = <div>Childrens content</div>;

describe(`PropertyHost`, () => {
  it(`should render PropertyHost correctly`, () => {
    const tree = renderer.create(
        <PropertyHost host={USERS[0]}>
          {CHILDREN}
        </PropertyHost>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
