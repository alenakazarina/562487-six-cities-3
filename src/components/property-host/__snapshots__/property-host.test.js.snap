// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyHost should render PropertyHost correctly 1`] = `
<div
  className="property__host"
>
  <h2
    className="property__host-title"
  >
    Meet the host
  </h2>
  <div
    className="property__host-user user"
  >
    <div
      className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
    >
      <img
        alt="Alena avatar"
        className="property__avatar user__avatar"
        height={74}
        src="..//user1_avatar"
        width={74}
      />
    </div>
    <span
      className="property__user-name"
    >
      Alena
    </span>
  </div>
  <div>
    Childrens content
  </div>
</div>
`;
